Snowflake.Data/Client/SnowflakeDbConnectionPool.cs

```diff
@@ -68,6 +68,7 @@ internal static SnowflakeDbConnection getConnection(string connStr)
             }
             lock (_connectionPoolLock)
             {
+                cleanExpiredConnections(); // clean up all expired connections first.
```

Comment:
Will this cause delay in getting a connection? If yes, then I suggest when connection string match in the pool, do one more check whether it's expired or not. If expired, then put in a separate list, say expire list and continue look into pool. Don't change pool count. After the loop, which are in expire list, remove those only. This way we remove expire connection in batches and not holding to get a connection from a pool to cleanup all expired connections.